test/tests/serial/open_dkim.bats

```diff
@@ -1,465 +1,288 @@
-  local PRIVATE_CONFIG
-  PRIVATE_CONFIG=$(duplicate_config_for_container . "${BATS_TEST_NAME}")
-  rm -rf "${PRIVATE_CONFIG}/with-selector" && mkdir -p "${PRIVATE_CONFIG}/with-selector"
-
-  # Generate first key
-  run docker run --rm \
-    -e LOG_LEVEL='trace' \
-    -v "${PRIVATE_CONFIG}/with-selector/":/tmp/docker-mailserver/ \
-    "${IMAGE_NAME:?}" /bin/sh -c "open-dkim keysize 2048 domain 'domain1.tld' selector mailer| wc -l"
+function __should_generate_dkim_key() {
+  local EXPECTED_LINES=${1}
+  local ARG_KEYSIZE=${2}
+  local ARG_DOMAINS=${3}
+  local ARG_SELECTOR=${4}
 
-  assert_success
-  assert_output 4
+  local DKIM_CMD='open-dkim'
+  [[ -n ${ARG_KEYSIZE}  ]] && DKIM_CMD+=" keysize ${ARG_KEYSIZE}"
+  [[ -n ${ARG_DOMAINS}  ]] && DKIM_CMD+=" domain '${ARG_DOMAINS}'"
+  [[ -n ${ARG_SELECTOR} ]] && DKIM_CMD+=" selector '${ARG_SELECTOR}'"
 
-  # Check keys for domain1.tld
-  run docker run --rm \
-    -e LOG_LEVEL='trace' \
-    -v "${PRIVATE_CONFIG}/with-selector/opendkim":/etc/opendkim \
-    "${IMAGE_NAME:?}" /bin/sh -c 'ls -1 /etc/opendkim/keys/domain1.tld/ | wc -l'
+  _run_in_container_bash "${DKIM_CMD}"
 
   assert_success
-  assert_output 2
+  _should_output_number_of_lines "${EXPECTED_LINES}"
+}
 
-  # Check key names with selector for domain1.tld
-  run docker run --rm \
-    -e LOG_LEVEL='trace' \
-    -v "${PRIVATE_CONFIG}/with-selector/opendkim":/etc/opendkim \
-    "${IMAGE_NAME:?}" /bin/sh -c "ls -1 /etc/opendkim/keys/domain1.tld | grep -E 'mailer.private|mailer.txt' | wc -l"
+function __should_have_expected_files() {
+  local EXPECTED_KEYSIZE=${1}
+  local DKIM_DOMAIN='localhost.localdomain'
+  local TARGET_DIR="/tmp/docker-mailserver/opendkim/keys/${DKIM_DOMAIN}"
+
+  # DKIM private key for signing, parse it to verify private key size is correct:
+  _run_in_container_bash "openssl rsa -in '${TARGET_DIR}/mail.private' -noout -text"
+  assert_success
+  assert_line --index 0 "RSA Private-Key: (${EXPECTED_KEYSIZE} bit, 2 primes)"
+
+  # DKIM record, extract public key (base64 encoded, potentially multi-line)
+  # - tail to exclude first line,
+  # - then sed to extract values within quoted lines, then remove `p=` from the start,
+  # - and finally echo to concatenate all lines into single string
+  # Next decode and parse it with openssl to verify public-key key size is correct:
+  _run_in_container_bash "echo \$( \
+    tail -n +2 '${TARGET_DIR}/mail.txt' \
+    | sed -nE -e 's/.*\"(.*)\".*/\1/p' \
+    | sed -e 's/^p=//' \
+  ) | openssl enc -base64 -d | openssl pkey -inform DER -pubin -noout -text
+  "
+  assert_success
+  assert_line --index 0 "RSA Public-Key: (${EXPECTED_KEYSIZE} bit)"
+
+  # Contents is for expected DKIM_DOMAIN and selector (mail):
+  _run_in_container cat "${TARGET_DIR}/mail.txt"
+  assert_output --regexp "; ----- DKIM key mail for ${DKIM_DOMAIN}$"
+}
 
-  assert_success
-  assert_output 2
+function __should_have_key_for_domain() {
+  local KEY_DOMAIN=${1}
+  local KEY_SELECTOR=${2:-'mail'}
```

Comment:
```suggestion
  local KEY_SELECTOR=${2:-mail}
```

Don't ask why, but it seems to be good practice to leave out the `''` when providing a default.